framework/web/UrlRule.php

```diff
@@ -256,6 +355,25 @@ public function parseRequest($manager, $request)
         }
         $matches = $this->substitutePlaceholderNames($matches);
 
+        // if routing succeeded (some rule matched), check whether pathInfo
+        //  was changed during canonization. If so, take the required action.
+        if ($pathInfoCanonized != $pathInfoOrig) {
+            switch ($this->canonize['action']) {
+                case 'redirect':
+                    $url = '/' . $pathInfoCanonized
+                        . (Yii::$app->request->queryString ? '?' . Yii::$app->request->queryString : '');
+                    die('redirect ' . $url);
+                    Yii::$app->getResponse()->redirect($url)->send();
+                    break;
```

Comment:
Well, the initial idea was about doing redirection by default because, for example, Wordpress, which is very common, adds slashes to all URLs in the content therefore all links to Yii websites are becoming 404.